.github/workflows/python-module.yml

```diff
@@ -0,0 +1,147 @@
+name: Build Python module
+
+on: [push]
+
+jobs:
+  build_lin:
+    runs-on: ubuntu-18.04
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-python@v1
+      with:
+        python-version: '3.x' # Version range or exact version of a Python version to use, using semvers version range syntax.
+        architecture: 'x64' # (x64 or x86)
+    - name: Build testing docker container
+      run: |
+        docker build -f python/Dockerfile -t testing .
+    - name: Build
+      run: |
+        docker run \
+          -v$(pwd):/app \
+          -w/app/python \
+          testing \
+          bash -c "
+            \$PYBIN/python setup.py install
+            \$PYBIN/python setup.py bdist_wheel -p manylinux1_x86_64"
+    - name: Test
+      run: |
+        docker run \
+          -v$(pwd):/app \
+          -w/app/python \
+          testing \
+          bash -c "\$PYBIN/pytest tests"
+    - name: Check formatting
+      run: |
+        docker run \
+          -v$(pwd):/app \
+          -w/app/python \
+          testing \
+          bash -c "\$PYBIN/black --check ."
+    - name: Try to import on different python versions
+      run: |
+        docker run \
+          -v$(pwd):/app \
+          testing \
+          bash -c "
+            for PYBIN in /opt/python/{cp34-cp34m,cp35-cp35m,cp36-cp36m,cp37-cp37m}/bin; do
+              \$PYBIN/pip install /app/python/dist/wkw-*-py3-none-manylinux1_x86_64.whl
+              \$PYBIN/python -c 'import wkw; print(wkw.Dataset)'
+            done"
+    - name: Publish
+      if: startsWith(github.event.ref, 'refs/tags')
+      run: |
+         echo "[distutils]" >> .pypirc
+         echo "index-servers = pypi" >> .pypirc
+         echo "[pypi]" >> .pypirc
+         echo "repository=https://upload.pypi.org/legacy/" >> .pypirc
+         echo "username=${{ secrets.PYPI_USERNAME }}" >> .pypirc
+         echo "password=${{ secrets.PYPI_PASSWORD }}" >> .pypirc
+   
+         docker run \
+           -v$(pwd):/app \
+           -v$(pwd)/.pypirc:/root/.pypirc \
+           -w/app/python \
+           testing \
+           bash -c "\$PYBIN/python setup.py bdist_wheel -p manylinux1_x86_64 upload -r pypi"
+
+  build_mac:
+    runs-on: macOS-10.14
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-python@v1
+      with:
+        python-version: '3.x' # Version range or exact version of a Python version to use, using semvers version range syntax.
+        architecture: 'x64' # (x64 or x86)
+    - name: Install Rust and lz4
+      run: |
+        brew install rustup-init lz4
+        rustup-init -y
+    - name: Build
+      run: |
+        cd python
+        export PATH=$PATH:$HOME/.cargo/bin
+        python setup.py install
+    - name: Test
+      run: |
+        cd python
+        python -c "import wkw"
+        pip install pytest
+        pytest tests
+    - name: Publish
+      if: startsWith(github.event.ref, 'refs/tags')
+      run: |
+        echo "[distutils]" >> $HOME/.pypirc
```

Comment:
Can this block be extracted into a reusable snippet? Invoking a bash script with this content probably doesn't work, due to the `${{ ... }}` string interpolation :thinking:

Reply:
I now use `twine` instead which uses env vars instead and is the new recommended way of publishing modules.